src/frontend/src/redux/system/reducer.ts

```diff
@@ -1,22 +1,26 @@
 import { LOG_IN, LOG_OUT, SystemState, SystemActionTypes } from './types';
 
 const initialState: SystemState = {
-  loggedIn: false
+  loggedIn: false,
+  isAdmin: false
 };
 
 export function systemReducer(
   state = initialState,
   action: SystemActionTypes
 ): SystemState {
+  console.log('action', action);
   switch (action.type) {
     case LOG_IN: {
       return {
-        loggedIn: true
+        loggedIn: true,
+        isAdmin: action.isAdmin
       };
     }
     case LOG_OUT: {
       return {
-        loggedIn: false
+        loggedIn: false,
+        isAdmin: false
```

Comment:
`isAdmin` should not exist for log out.

Reply:
I'm not sure how to resolve this issue. Tried to remove it bu received an error: "Property 'isAdmin' is missing in type '{ loggedIn: false; }' but required in type 'SystemState'."

Reply:
https://github.com/codeforpdx/recordexpungPDX/pull/783/files#diff-076570d447b7ed9a8929b014215fae08R3 makes it required. You could change it to `isAdmin?: boolean;`. If for whatever reason that doesn't work, then keeping line 22 above is fine but we probably should add it to the types at https://github.com/codeforpdx/recordexpungPDX/pull/783/files#diff-076570d447b7ed9a8929b014215fae08R14 too.

Reply:
It looks to me like `isAdmin` can't be made optional here so long as it's a required field in SystemState (which is good) , so apart from Kent's alternate suggestion here this PR looks good.

Reply:
If only I saw this comment three minutes ago. Oh well, I already changed it to reflect Kent's suggestion, and I checked that the log in and log out were still working properly.